client/components/domains/register-domain-step/index.jsx

```diff
@@ -31,19 +32,53 @@ var wpcom = require( 'lib/wp' ).undocumented(),
 	abtest = require( 'lib/abtest' ).abtest;
 
 // max amount of domain suggestions we should fetch/display
-var SUGGESTION_QUANTITY = 10,
+const SUGGESTION_QUANTITY = 10,
 	INITIAL_SUGGESTION_QUANTITY = 2;
 
+const analytics = analyticsMixin( 'registerDomain' );
+
+let searchQueue = [],
+	searchStackTimer = null;
+
+function processSearchQueue() {
+	var q = searchQueue.slice();
+	window.clearTimeout( searchStackTimer );
+	searchStackTimer = null;
+	searchQueue = [];
+
+	outerLoop:
+		for ( let i = 0; i < q.length; i++ ) {
+			for ( let k = i + 1; k < q.length; k++ ) {
+				if ( startsWith( q[ k ].query, q[ i ].query ) ) {
+					continue outerLoop;
+				}
+			}
+			reportSearch( q[ i ] );
+		}
+}
+
+function reportSearch( { query, section } ) {
+	analytics.recordEvent( 'searchFormSubmit', query, section );
+}
+
+function enqueueSearch( search ) {
+	searchQueue.push( search );
+	if ( searchStackTimer ) {
+		window.clearTimeout( searchStackTimer );
+	}
+	searchStackTimer = window.setTimeout( processSearchQueue, 10000 );
+}
+
 var RegisterDomainStep = React.createClass( {
-	mixins: [ analyticsMixin( 'registerDomain' ) ],
+	mixins: [ analytics ],
 
 	propTypes: {
 		cart: React.PropTypes.object,
 		onDomainsAvailabilityChange: React.PropTypes.func,
 		products: React.PropTypes.object.isRequired,
 		selectedSite: React.PropTypes.oneOfType( [ React.PropTypes.object, React.PropTypes.bool ] ),
 		basePath: React.PropTypes.string.isRequired,
-		suggestion: React.PropTypes.string,
+		suggestion: React.PropTypes.string
```

Comment:
The reason I'm against having trailing commas is JavaScript is broken when it comes to that. It's an old convention to remove commas and I would like to stick with it. See here: http://stackoverflow.com/questions/7246618/trailing-commas-in-javascript for more context.

Also while it produces a tiny bit cleaner diff, it shouldn't make the review harder as a whole. And it is not easier because the amount of keystrokes are the same.